lib/addons/apache-airflow/index.ts

```diff
@@ -0,0 +1,237 @@
+ */
+export class AirflowAddOn extends HelmAddOn {
+
+    readonly options: AirflowAddOnProps;
+
+    constructor(props?: AirflowAddOnProps) {
+        super({...defaultProps, ...props});
+        this.options = this.props as AirflowAddOnProps;
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+
+        // Create Namespace
+        const ns = createNamespace(this.options.namespace!, cluster, true, true);
+
+        let values: Values = populateValues(clusterInfo, ns, this.options);
+
+        const chart = this.addHelmChart(clusterInfo, values, false, false);
+
+        return Promise.resolve(chart);
+    }
+}
+
+/**
+ * populateValues populates the appropriate values used to customize the Helm chart
+ * @param helmOptions User provided values to customize the chart
+ */
+function populateValues(clusterInfo: ClusterInfo, ns: KubernetesManifest, helmOptions: AirflowAddOnProps): Values {
+    const values = helmOptions.values ?? {};
+
+    const albAddOnCheck = clusterInfo.getScheduledAddOn(AwsLoadBalancerControllerAddOn.name);
+    const cert = helmOptions.certificateResourceName;
+    const dbConfig = helmOptions.dbConfig;
+    const bucket = helmOptions.s3BucketName;
+    const efs = helmOptions.efsFileSystemName;
+
+    // Kubernetes settings
+    setPath(values, "config.kubernetes.namespace", helmOptions.namespace!);
+    setPath(values, "config.kubernetes_executor.namespace", helmOptions.namespace!);
+
+    // Security Context
+    setPath(values, "securityContext.fsGroup", 66534);
+    setPath(values, "executor", "KubernetesExecutor");
+
+    // Load Balancer
+    if (helmOptions.enableAlb){
+        assert(albAddOnCheck, `Missing a dependency: ${AwsLoadBalancerControllerAddOn.name}. Please add it to your list of addons.`); 
+        const presetAnnotations: any = {
+            'alb.ingress.kubernetes.io/group.name': 'airflow',
+            'alb.ingress.kubernetes.io/scheme': 'internet-facing',
+            'alb.ingress.kubernetes.io/target-type': 'ip',
+            'alb.ingress.kubernetes.io/listen-ports': '[{"HTTP": 80}]',
+            'alb.ingress.kubernetes.io/healthcheck-path': '/health',
+        };
+        if (helmOptions.certificateResourceName){
+            presetAnnotations['alb.ingress.kubernetes.io/listen-ports'] = '[{"HTTP": 80},{"HTTPS":443}]';
+            const certificate = clusterInfo.getResource<ICertificate>(helmOptions.certificateResourceName);
+            presetAnnotations['alb.ingress.kubernetes.io/certificate-arn'] = certificate?.certificateArn;
+        } 
+        
+        setPath(values, "ingress.web", {
+            "enabled": "true",
+            "annotations": presetAnnotations,
+            "pathType": "Prefix",
+            "ingressClassName": "alb",
+        });
+    } else {
+        assert(!cert, 'Cert option is supported only if ALB is enabled.');
+    }
+
+    // Using RDS as a Database
+    if (helmOptions.enableRds){
```

Comment:
The flow is a bit hard to read, code is [not clean](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882).

Can you introduce a few methods to reduce the size and avoid too much branching? 

can it work if enableRds is false and no dbconfig provided?